pkg/pgmodel/ingestor/trace/writer.go

```diff
@@ -324,6 +327,13 @@ func (t *traceWriterImpl) InsertTraces(ctx context.Context, traces pdata.Traces)
 
 	// Only report telemetry if ingestion successful.
 	tput.ReportSpansProcessed(timestamp.FromTime(time.Now()), traces.SpanCount())
+
+	// since otel is making Protobufs internal this is our only chance to get the size of the message
+	tracesSizer, ok := tracesMarshaller.(pdata.TracesSizer)
```

Comment:
Can we benchmark this vs using binary.BigEndian and taking length of the output? Just want to make sure that size calculation does not take a hit on perf